ompi/mca/mtl/ofi/Makefile.am

```diff
@@ -26,7 +44,21 @@ mtl_ofi_sources = \
 				   mtl_ofi_endpoint.h \
 				   mtl_ofi_endpoint.c \
 				   mtl_ofi_request.h \
-				   mtl_ofi_types.h
+				   mtl_ofi_types.h \
+				   mtl_ofi_opt.h \
+				   $(generated_sources)
+
+# A number of files are generated from macro expansion to minimize
+# branches in the critical path.  These files have perl modules with the suffix
+# .pm that generate the corresponding .c file with all possible branches as
+# their own function and symbol. Additional input
+# files should be added to generated_source_modules, as well as adding
+# their .c variants to generated_sources.
+%.c : %.pm;
+	$(PERL) generate-opt-funcs.pl $@
+
+clean:
```

Comment:
I had added this due to the fact that git clone development showed that if a generic function was changed the specialized file would no longer compile. This clean target made it easy to resolve the issue. However, if this overrides the automake clean I will remove this, thanks!

Reply:
I had added it due to the fact that during git clone development if one changed one of the generic functions that specialized file would no longer compile and the clean allowed for the file to be easily removed by the makefile target. However, if that is not allowed and overrides the maintainer clean I can remove that target. Thanks!

Reply:
Yes, it will override the clean target (and not clean the other things).  But almost worse than that, it will remove the generated source files on a "make clean", which means customers would need to have perl as a build time dependency.  Since we only require Perl of maintainers (ie, those who run autogen) and not users (ie, those who run configure/make), that would break the customer experience.

Can you walk me through the use case that caused problems a bit more?  It sounds like we missed a dependency we should express, but not sure I see the problem well enough to offer advice on how to fix it.

Reply:
I rechecked the scenario and ended up removing the clean target. As long as the MAINTAINERCLEAN is set, then it is not necessary.